backend/src/packages/sites/site.service.ts

```diff
@@ -40,18 +44,66 @@ class SiteService implements Omit<IService, 'find' | 'update' | 'delete'> {
   public async create(
     payload: SiteCreateRequestDto,
   ): Promise<SiteCreateResponseDto> {
-    const site = await this.siteRepository.create(
-      SiteEntity.initializeNew(payload),
-    );
+    const site = await this.siteRepository
+      .create(SiteEntity.initializeNew({ name: payload.name }))
+      .then((site) => site.toObject());
```

Comment:
```suggestion
    const entity = await this.siteRepository
      .create(SiteEntity.initializeNew({ name: payload.name }))
   const site = rawSite.toObject()
```
prefer 1 approach for working with async